raft-tech/TANF-app
# Cat1 Cleanup

## Summary of Changes
- Updated based on error audit language
- Added code infrastructure that will allow smoother implementation of cat2/cat3 error audit updates
Pull request closes #2673 

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose up
cd tdrs-backend && docker-compose up
```

- Set `GENERATE_TRAILER_ERRORS=True/False` in your compose file for the web app depending on how you're testing
- Submit one of the bad trailer datafiles and see that the parser errors are or arent generated based on your setting from above
- Verify logs for trailer errors always exist regardless of setting
- Verify new error messages on Cat1 validators lineup with Error Audit HackMD.

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] All trailer length issues should be ignored / omitted from error reports and parsing requirements 
- [x] Cat 1 errors should be consistent with Desired Outputs in the [Cat 1 Validators section](https://hackmd.io/itv2dcDPRVulWm5mcaJtHQ?view#Cat-1-Validators) of the errors audit 
- [x] record type (T1, T2, M1, etc) appears in every row of the parsing error report download

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
